corehq/project_limits/rate_limiter.py

```diff
@@ -117,53 +124,59 @@ def get_n_users_for_rate_limiting(domain):
 
     """
-    n_users = _get_user_count(domain)
-    n_users_included_in_subscription = _get_users_included_in_subscription(domain)
+    n_users_included_in_subscription = _get_users_included_in_subscription(domain, enterprise_limit)
+    if enterprise_limit:
+        n_users = 0
+    else:
+        n_users = _get_user_count(domain)
     return max(n_users_included_in_subscription, n_users)
 
 
 def _get_user_count(domain):
     return CommCareUser.total_by_domain(domain, is_active=True)
 
 
-def _get_users_included_in_subscription(domain):
+def _get_users_included_in_subscription(domain, enterprise_limit):
     from corehq.apps.accounting.models import Subscription
     subscription = Subscription.get_active_subscription_by_domain(domain)
     if subscription:
         plan_version = subscription.plan_version
-
-        n_included_users = (
-            plan_version.feature_rates.get(feature__feature_type='User').monthly_limit)
-
-        if plan_version.plan.is_customer_software_plan:
-            # For now just give each domain that's part of an enterprise account
-            # access to nearly all of the throughput allocation.
-            # Really what we want is to limit enterprise accounts' submissions accross all
-            # their domains together, but right now what we care about
-            # is not unfairly limiting high-paying enterprise accounts.
-            n_domains = len(plan_version.subscription_set.filter(is_active=True))
-            # Heavily bias towards allowing high throughput
-            # 80% minimum, plus a fraction of 20% inversely proportional
-            # to the number of domains that share the throughput allocation.
-            return n_included_users * (.8 + .2 / n_domains)
+        num_users = plan_version.feature_rates.get(feature__feature_type='User').monthly_limit
+        if enterprise_limit or not plan_version.plan.is_customer_software_plan:
+            return num_users
         else:
-            return n_included_users
+            return 0
     else:
         return 0
 
 
+@quickcache(['domain'], memoize_timeout=60, timeout=60 * 60)
+def _get_account_name(domain):
+    from corehq.apps.accounting.models import BillingAccount
+    return BillingAccount.get_account_by_domain(domain).name
+
+
 class PerUserRateDefinition(object):
     def __init__(self, per_user_rate_definition, constant_rate_definition=None):
         self.per_user_rate_definition = per_user_rate_definition
         self.constant_rate_definition = constant_rate_definition or RateDefinition()
 
     def get_rate_limits(self, domain):
-        n_users = get_n_users_for_rate_limiting(domain)
-        return (
-            self.per_user_rate_definition
-            .times(n_users)
-            .plus(self.constant_rate_definition)
-        ).get_rate_limits()
+        domain_users = get_n_users_for_rate_limiting(domain)
+        enterprise_users = get_n_users_for_rate_limiting(domain, enterprise_limit=True)
```

Comment:
If you were to continue along the lines I outlined above, these lines would then be something like

```python
        domain_users = get_n_users_in_domain(domain)
        subscription_users = get_n_users_in_subscription(domain)
```